templates/tfengine/components/audit/main.tf

```diff
@@ -40,84 +40,67 @@ resource "google_{{.parent_type}}_iam_audit_config" "config" {
-  location                    = "{{.bigquery_location}}"
-  default_table_expiration_ms = 365 * 8.64 * pow(10, 7) # 365 days
-  access = [
-    {
-      role          = "roles/bigquery.dataOwner",
-      special_group = "projectOwners"
-    },
-    {
-      role           = "roles/bigquery.dataViewer",
-      group_by_email = var.auditors_group
-    },
-  ]
+  source  = "terraform-google-modules/log-export/google//modules/bigquery"
+  version = "~> 5.0.0"
+
+  dataset_name             = "{{.logs_bigquery_dataset.dataset_id}}"
+  project_id               = module.project.project_id
+  location                 = "{{.bigquery_location}}"
+  log_sink_writer_identity = "${module.bigquery_export.writer_identity}"
+  expiration_days          = 365
 }
 
-resource "google_project_iam_member" "bigquery_sink_member" {
-  project = module.bigquery_destination.bigquery_dataset.project
-  role    = "roles/bigquery.dataEditor"
-  member  = google_logging_{{.parent_type}}_sink.bigquery_audit_logs_sink.writer_identity
-}
-
-# Cloud Storage log sink.
-resource "google_logging_{{.parent_type}}_sink" "storage_audit_logs_sink" {
-  name                 = "storage-audit-logs-sink"
-  {{$parent_field}}    = {{$parent_var}}
-  include_children     = true
-  filter               = "{{$filter}}"
-  destination          = "storage.googleapis.com/${module.storage_destination.bucket.name}"
+module "storage_export" {
+  source  = "terraform-google-modules/log-export/google"
+  version = "~> 5.0.0"
+
+  destination_uri        = "${module.storage_destination.destination_uri}"
+  filter                 = "{{$filter}}"
+  log_sink_name          = "storage-audit-logs-sink"
+  parent_resource_type   = "{{.parent_type}}"
+  parent_resource_id     = {{$parent_var}}
+  unique_writer_identity = true
+  include_children       = true
 }
 
 module "storage_destination" {
-  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
-  version = "~> 1.7.0"
-
-  name          = "{{.logs_storage_bucket.name}}"
-  project_id    = module.project.project_id
-  location      = "{{.storage_location}}"
-  storage_class = "COLDLINE"
-
-  lifecycle_rules = [{
-    action = {
-      type = "Delete"
-    }
-    condition = {
-      age        = 7 * 365 # 7 years
-      with_state = "ANY"
-    }
-  }]
-
-  iam_members = [
-    {
-      role   = "roles/storage.objectViewer"
-      member = "group:${var.auditors_group}"
-    },
-  ]
+  source  = "terraform-google-modules/log-export/google//modules/storage"
+  version = "~> 5.0.0"
+
+  storage_bucket_name      = "{{.logs_storage_bucket.name}}"
+  project_id               = module.project.project_id
+  location                 = "{{.storage_location}}"
+  log_sink_writer_identity = "${module.storage_export.writer_identity}"
+  storage_class            = "COLDLINE"
+  expiration_days          = 7*365
+  retention_policy = {
+    is_locked             = true
+    retention_period_days = 6 * 365
```

Comment:
what's the different between the `retention_period_days` here and `expiration_days` above? and why are they set to different values?

Reply:
Retention period does not actually do anything. It just forces the bucket to permanently have an expiry that is at least this amount. See https://cloud.google.com/storage/docs/bucket-lock. I also added a comment.

p.s. there seems to be an org policy we should probably set as well: https://cloud.google.com/storage/docs/org-policy-constraints#audit-logging. Do you want to look into that?

Reply:
sure i'll look into that.